### PR TITLE
Make ObjectPermission a Changelogged model.

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #4992 
# What's Changed

Screen Shot 2024-01-02 at 2 10 31 PM
Screen Shot 2024-01-02 at 2 09 45 PM
Screen Shot 2024-01-02 at 2 10 04 PM
Screen Shot 2024-01-02 at 2 10 21 PM

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/4992.changed [hunk 1]
```diff
@@ -0,0 +1 @@
+Make ObjectPermission a Changelogged model.
```
Comment (glennmatthews) [inline changes/4992.changed:1]:
```suggestion
Added change-logging (ObjectChange support) for the ObjectPermission model.
```

File: nautobot/docs/user-guide/platform-functionality/users/objectpermission.md [hunk 2]
```diff
@@ -54,3 +54,7 @@ To achieve a logical OR with a different set of constraints, define multiple obj
 ```
 
 Additionally, where multiple permissions have been assigned for an object type, their collective constraints will be merged using a logical "OR" operation.
+
++++ 2.1.1
+
+The ObjectPermission model now has changelogging capabilities. If an admin user modifies ObjectPermission instances from the django admin page, change logs will be automatically generated and will be accessible from the Nautobot main menu > Extensibility > Change Log.
```
Comment (glennmatthews) [inline nautobot/docs/user-guide/platform-functionality/users/objectpermission.md:60]:
```suggestion
+++ 2.1.1
    The ObjectPermission model now has change-logging capabilities. When object permissions are created, updated, or deleted, change logs will be automatically generated and will be viewable by users with the appropriate permissions.
```

File: nautobot/users/migrations/0008_make_object_permission_a_changelogged_model.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    operations = [
+        migrations.AddField(
+            model_name="objectpermission",
+            name="created",
+            field=models.DateTimeField(auto_now_add=True, null=True),
+        ),
+        migrations.AddField(
+            model_name="objectpermission",
+            name="last_updated",
+            field=models.DateTimeField(auto_now=True, null=True),
```
Comment (glennmatthews) [inline nautobot/users/migrations/0008_make_object_permission_a_changelogged_model.py:20]: For existing ObjectPermissions, I'm presuming these fields are given a null value on migration automatically, so we don't need a data migration? Just confirming.

**Comment by gsnider2195** [comment]
Do we want to add `get_absolute_url` to `ObjectPermission` so it can be linked in the `ObjectChange` list view?

**Comment by HanlinMiao** [comment]
> Do we want to add `get_absolute_url` to `ObjectPermission` so it can be linked in the `ObjectChange` list view?

I do not think there is a detail view `ObjectPermission` in Django admin page. As far as I can tell, `ObjectPermission` only has an update view. Should I make the `get_absolute_url` point to the update view?

File: nautobot/users/migrations/0008_make_object_permission_a_changelogged_model.py (hunk 3 above)
Comment (HanlinMiao) [reply nautobot/users/migrations/0008_make_object_permission_a_changelogged_model.py:20]: These fields are actually auto-populated with the timestamp when I ran the migration.
```
# Django version 3.2.23
# Nautobot version 2.1.1b1
# Example Nautobot App version 1.0.0
Python 3.10.13 (main, Dec 19 2023, 03:47:28) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> obj = ObjectPermission.objects.first()
>>> obj.last_updated
datetime.datetime(2024, 1, 2, 22, 7, 48, 950988, tzinfo=<UTC>)
>>> obj.created
datetime.datetime(2024, 1, 2, 22, 7, 1, 489508, tzinfo=<UTC>)
>>> obj = ObjectPermission.objects.last()
>>> obj.last_updated
datetime.datetime(2024, 1, 2, 22, 7, 1, 552173, tzinfo=<UTC>)
>>> obj.created
datetime.datetime(2024, 1, 2, 22, 7, 1, 489508, tzinfo=<UTC>)
>>> 
```

**Comment by gsnider2195** [comment]
> > Do we want to add `get_absolute_url` to `ObjectPermission` so it can be linked in the `ObjectChange` list view?
> 
> I do not think there is a detail view for `ObjectPermission` in Django admin page. As far as I can tell, `ObjectPermission` only has an update view. Should I make the `get_absolute_url` point to the update view?

Do we link to admin objects anywhere outside of the admin views? We don't hyperlink the user in the change logging views. It's probably fine to leave it the way it is.

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**